### PR TITLE
Solver code

### PR DESCRIPTION
**Pull request by Rifat8080**
In this project we :

Create a class called Solver
tests all of the methods described bellow by using Rspec and TDD
Create a method called factorial
Create a method called reverse
Create a method called fizzbuzz

### CODE REVIEW

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Solver
```
Comment (DammyShittu) [inline solver.rb:1]: - [ ] From your commit history you added the class and methods before writing the tests, but you were supposed to write the tests before adding the class, as that is the purpose of the project. Kindly do that.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Rifat8080 and @jubaer919,

## Highlights
✔️ Edge case testing

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Tested Edge cases
✅ All tests are passing 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
